pipeline_dp/budget_accounting.py

```diff
@@ -299,18 +301,30 @@ def _compose_distributions(
         composed, pld = None, None
 
         for mechanism_spec_internal in self._mechanisms:
-            if mechanism_spec_internal.mechanism_spec.noise_kind == NoiseKind.LAPLACE:
+            if mechanism_spec_internal.mechanism_spec.mechanism_type == MechanismType.LAPLACE:
                 # The Laplace distribution parameter = std/sqrt(2).
                 pld = pldlib.PrivacyLossDistribution.from_laplace_mechanism(
                     mechanism_spec_internal.sensitivity *
                     noise_standard_deviation / math.sqrt(2) /
                     mechanism_spec_internal.weight,
                     value_discretization_interval=self._pld_discretization)
-            elif mechanism_spec_internal.mechanism_spec.noise_kind == NoiseKind.GAUSSIAN:
+            elif mechanism_spec_internal.mechanism_spec.mechanism_type == MechanismType.GAUSSIAN:
                 pld = pldlib.PrivacyLossDistribution.from_gaussian_mechanism(
                     mechanism_spec_internal.sensitivity *
                     noise_standard_deviation / mechanism_spec_internal.weight,
                     value_discretization_interval=self._pld_discretization)
+            elif mechanism_spec_internal.mechanism_spec.mechanism_type == MechanismType.GENERIC:
+                # It is required to convert between the noise_standard_deviation of a Laplace or Gaussian mechanism
+                # and the (epsilon, delta) Generic mechanism because the calibration is defined by one parameter.
+                # There are multiple ways to do this; here it is assumed that (epsilon, delta) specifies the Laplace
+                # mechanism and epsilon is computed based on this. The delta is computed to be proportional to epsilon.
+                epsilon_0 = math.sqrt(2) / noise_standard_deviation
+                delta_0 = epsilon_0 / self._total_epsilon * self._total_delta
+                mechanism_spec_internal.mechanism_spec.set_eps_delta(
```

Comment:
@dvadym 👋 Adding this call to set epsilon and delta into the mechanism for the generic mechanism type which is checked in the tests.

Reply:
Please put this line in `compute_budgets()` inside loop
`for mechanism in self._mechanisms:
  ...`

Reply:
👍 @dvadym I moved these outside of the binary search function, so they are set once outside in `compute budgets`. To try to avoid confusion, I renamed those in `_compose_distributions` to `_interim`, if that makes sense. Please let me know if it does not - thank you!